### DISCUSSION TITLE
Ruby: World-writable directory in PATH.

### DISCUSSION THREAD

**Discussion by matthewtodd** [community/community#25801] [Actions]
<p>I’m running <code>ruby -w</code> in my build script, which is producing <a href="https://github.com/matthewtodd/workspace/commit/fe2dbc90e69f1835abb4ef911eb58e981e28073c/checks?check_suite_id=300434236" rel="nofollow noopener">warnings</a> about /opt/hostedtoolcache/Ruby/2.6.3/x64/bin being world-writable. (Indeed, poking around reveals that all of its parent directories are, too.)</p>
<p>Even if mine is the only process in the virtual environment (mitigating any practical security concerns), I’d still love to keep the build warning-free. I’m currently working around this one by running <code>env -i /usr/bin/ruby -w</code>, but it would be nice to see it fixed upstream.</p>
<p>Thoughts? Thank you!</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi Matthew,</p>
<p>Thank you for taking time to report this!</p>
<p>This warning is due to the directory has been added to your $PATH(the list of directories the OS searches when trying to find an executable to launch),  and permission is writable anyone can write to. This is potential a security problem, Ruby notices this and issues the warning.</p>
<p>However,  the hosted-github runner is a temporary virtual environment which will be deprecated after workflow run complete.  GitHub hosts Linux and Windows runner is a fork of azure pipeline agent, it’s safe and protected. In addition, make the directory in $PATH un-writable by default will cause other problems.</p>
<p>As a workaround, please use below command to change the permissions of the directory so that it is no longer world writable.</p>
<p>chmod go-w /directory</p>
<p>Thanks.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/d/d0a4709ffb8959aa652286f9b9a494fe0d8659cc.png" data-download-href="/uploads/short-url/tLJCJJV0KlIf93rHlgjrcWtd7Jy.png?dl=1" title="Ruby.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181093544-4bc8c0e0-6e06-42dd-ba8e-792cfe1dd013.png" alt="Ruby.png" data-base62-sha1="tLJCJJV0KlIf93rHlgjrcWtd7Jy" width="690" height="390" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/d/d0a4709ffb8959aa652286f9b9a494fe0d8659cc_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Ruby.png</span><span class="informations">1073×607 30.7 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by matthewtodd** [comment]
<p>Thanks, <a class="mention" href="/weide-zhou">@weide-zhou</a>, I appreciate your taking the time to respond, and I’m happy to hear this will only be an issue during the beta period.</p>
<p>I had previously <a href="https://github.com/matthewtodd/workspace/commit/472a0ba808f77e96edc599d55790a252f8e3ee3d/checks?check_suite_id=300453460" rel="nofollow noopener">tried</a> the chmod route, but I didn’t think to try sudo. Good to know <a href="https://xkcd.com/149/" rel="nofollow noopener">that’ll work</a>!</p>
